views_api.py

```diff
@@ -151,20 +157,25 @@ async def api_submarineswap_create(data: CreateSubmarineSwap):
             status_code=HTTPStatus.METHOD_NOT_ALLOWED,
             detail="auto reverse swap is active, a swap would immediatly be swapped out again.",
         )
-
-    client = create_boltz_client()
-    swap_id = urlsafe_short_hash()
-    payment_hash, payment_request = await create_invoice(
-        wallet_id=data.wallet,
-        amount=data.amount,
-        memo=f"swap of {data.amount} sats on boltz.exchange",
-        extra={"tag": "boltz", "swap_id": swap_id},
-    )
-    refund_privkey_wif, swap = client.create_swap(payment_request)
-    new_swap = await create_submarine_swap(
-        data, swap, swap_id, refund_privkey_wif, payment_hash
-    )
-    return new_swap.dict() if new_swap else None
+    try:
+        client = create_boltz_client()
+        swap_id = urlsafe_short_hash()
+        payment_hash, payment_request = await create_invoice(
+            wallet_id=data.wallet,
+            amount=data.amount,
+            memo=f"swap of {data.amount} sats on boltz.exchange",
+            extra={"tag": "boltz", "swap_id": swap_id},
+            expiry=60*60*24, # 1 day
+        )
+        refund_privkey_wif, swap = client.create_swap(payment_request)
+        new_swap = await create_submarine_swap(
+            data, swap, swap_id, refund_privkey_wif, payment_hash
+        )
+        return new_swap.dict() if new_swap else None
+    except Exception as exc:
+        raise HTTPException(
+            status_code=HTTPStatus.METHOD_NOT_ALLOWED, detail=str(exc)
```

Comment:
`METHOD_NOT_ALLOWED` seems like the wrong error to throw for an internal server error